### ISSUE TITLE
[Firefox] about:config suggestion: privacy.resistFingerprinting.letterboxing;true

### ISSUE THREAD

**Issue by Mikaela** [privacytoolsIO/privacytools.io#1630]
`privacy.resistFingerprinting.letterboxing` is introduced in Firefox 67 (https://github.com/privacytoolsIO/privacytools.io/issues/1006#issuecomment-504778518) and brings [the bars around resized Tor Browser](https://support.torproject.org/#maximized-torbrowser-window) to Firefox hiding the real screen size from websites. **The boolean option has to be created in `about:config` by the user.**

Personally I like the option and use it to also get used to the bars existing on Tor Browser, but I am not entirely sure if we should list it, so I wish for feedback.

I don't know if there is a benefit from hiding the screen size from websites (unless Tor Browser is being used) as the user is fingerprintable otherwise and I have heard an argument that *window rounded to a multiple of 200px x 100px* seems weird or unusual coming from someone who is not using Tor (Browser). I also don't know how widespread tracking the screen size is or if it's in any threat model outside of Tor Browser.

On the other hand us listing it could affect how common or not the option is?

See also: [MEDIA WINDOW SIZE POC](https://demos.traudt.xyz/css/media/) which changes the numbers as the window is resized, and it's [companion post](https://matt.traudt.xyz/p/YF4ciVY6.html) (note: they are written with Tor Browser in mind).

**Comment by Thorin-Oakenpants** [comment]
Hey Mikaela 

Use [this](https://ghacksuserjs.github.io/TorZillaPrint/TorZillaPrint.html#screen) and resize, zoom as much as you want and watch the auto-magic (for desktop, not android)

---

My ~~18.25 cents~~ ~~$2~~ $6.40 worth...

When someone gets moving on #1430 (been sitting idle for 2 months waiting for commits) ...**then** there will be (I envisage) 3 sections: no-breakage, some-breakage/annoyance, and here-be-dragons - and the list is limited and advanced options we point to the ghacks user.js

Without checking ... `privacy.resistFingerprinting` (RFP) is listed. Letterboxing (LB) is pointless without RFP. **So you would put the two together**. Using LB without RFP would just be pointless. But let's at least do first things first ... #1430 

---

on LBing

The idea *was* to enforce LB with RFP (whether the new window sizes on open would remain is debatable ) - but for now, it's not part of RFP but a separate option, and it looks like it will remain that way - as optional (with a UI checkbox) for TB (and also upstreamed to Firefox). What should happen, IMO, is LBing should only be applied if RFP is enabled

Anyway: yes, I would add it as a optional **companion pref** to RFP.  LBing protects inner window measurements. On it's own it's pointless (you're already leaking so much) or it's actually harmful/unique  (FPing likes stability). It all depends on how you use your Firefox. Additionally, your threat model comes into it.

---

on FPing...

You control this in several layers: first by blocking fingerprinters (known scripts) and 3rd party requests. This stops most of it. Secondly, you mitigate those that are missed (e.g. maybe some 1st party scripts) by breaking web standards: i.e spoofing, randomizing, whatever. Lastly, you reduce the attack surface for what is left: i.e. disable the API. There's more to it than this, but I'm keeping it simple.

With or without RFP, with or without LBing, everyone is already highly FP'able in Firefox. RFP is not even close to being a be-all-end-all solution. And Firefox does not have a enforced RFP set of users (e.g all PB mode users). So 1: not enforced (but the set could be quite large, but I don't think it would be as large as TB's 6 million or whatever it is - IDK) and 2: not enough on it's own (e.g disabled APIs etc). And .. just to give you all a downer .. even then, there are FPing holes not covered (and I won't discuss them)

tl;dr: you're all screwed if FPing gets through and digs deep enough. Your [best defence](https://en.wikipedia.org/wiki/The_best_defense_is_a_good_offense) is blocking the scripts in the first place. And your last defence relies on Firefox integrating a `Super Private Browsing Mode` that utilizes Tor and adds all the tor project protections. And that last defence won't happen if Tor can't scale

But do cheer up, at least you're alive :)

**Comment by Mikaela** [comment]
@Thorin-Oakenpants thanks for your comment (while I wonder why (presumably) you removed it), I guess I am closing this in favour of #1430.
